### PR TITLE
Add COSPAD to QMK and add functions to read HSV

### PR DESCRIPTION
**Pull request by kamisamamizu**
I have added three functions to rgblight.c to be able to read the hue, saturation and value from other parts of the code.
They are rgblight_get_hue(), rgblight_get_sat(), adn rgblight_get_val().

### CODE REVIEW

File: keyboards/cospad/rules.mk [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#   Teensy halfKay   512
+#   Teensy++ halfKay 1024
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/cospad/rules.mk]: This isn't really used anymore.  It should be `BOOTLOADER = caterina` or the like, instead. 

That will set the size and some other options, as well as special handling, it needed. 

This is more of a nitpick/consistency thing for me.  So feel free to ignore or what not! :)

Comment (kamisamamizu) [reply keyboards/cospad/rules.mk]: I used amjpad as a base for the port, as it also is a 6x4 numpad. And it seems it hasn't been updated, from what you tell me.
It's an easy change, however, so I can make the changes easily.

Comment (jackhumbert) [reply keyboards/cospad/rules.mk]: This is fine for now - we have lots of these we need to update :) thanks @drashna!

**Comment by jackhumbert** [comment]
Awesome - thanks! :)

**Pull request merged**